src/journal/journald-kmsg.c

```diff
@@ -325,14 +332,6 @@ static int server_read_dev_kmsg(Server *s) {
         if (l == 0)
                 return 0;
         if (l < 0) {
-                /* Old kernels which don't allow reading from /dev/kmsg return EINVAL when we try. So handle
-                 * this cleanly, but don't try to ever read from it again. */
-                if (errno == EINVAL) {
-                        s->dev_kmsg_event_source = sd_event_source_unref(s->dev_kmsg_event_source);
-                        s->dev_kmsg_readable = false;
```

Comment:
This can be dropped since nothing would set it to false anymore? Simple check of dev_kmsg_fd is sufficient?